src/Onnx/OnnxToFeld.hs

```diff
@@ -236,42 +264,142 @@ mkInitReadFile hf initGroups = L.concat $ start : map initVec initGroups ++ map
             where t = fromString $ renderType $ tiToType ti
                   field = show (tiField ti + 1)
         initVec [] = error "Impossible"
-        weightRecTC = fromString $ renderType $ mkStructType $ map toFieldType initGroups
-        toFieldType (ti:_) = ("member" ++ show (tiField ti + 1), mkAwLType universal $ tiToType ti)
-        toFieldType []     = error "Impossible"
+
+-- | Construct a main program
+mkMainFile :: FilePath -> FilePath -> FilePath -> L.ByteString -> [V.ValueInfoProto] -> [V.ValueInfoProto]-> L.ByteString
+mkMainFile bname hf weightFile weightRecTC outputs inputs = start
+  where start = L.unlines $
+                [ "#include \"" <> fromString hf <> "\""
+                , ""
+                , "#include <stdlib.h>"
+                , "#include <stdio.h>"
+                , "#include <inttypes.h>"
+                , ""
+                , "static void check(int n, int ref)"
+                , "{"
+                , "  static int items = 0;"
+                , "  if( n != ref ) {"
```

Comment:
Same style comments for this code as the previous check().

Reply:
Ok.